src/JSInterop/Microsoft.JSInterop/src/DotNetObjectRefManager.cs

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Threading;
+
+namespace Microsoft.JSInterop
+{
+    internal class DotNetObjectRefManager
+    {
+        private long _nextId = 0; // 0 signals no object, but we increment prior to assignment. The first tracked object should have id 1
+        private readonly ConcurrentDictionary<long, IDotNetObjectRef> _trackedRefsById = new ConcurrentDictionary<long, IDotNetObjectRef>();
+
+        public static DotNetObjectRefManager Current
+        {
+            get
+            {
+                if (!(JSRuntime.Current is JSRuntimeBase jsRuntimeBase))
+                {
+                    throw new InvalidOperationException("JSRuntime must be set up correctly and must be an instance of JSRuntimeBase to use DotNetObjectRef.");
+                }
+
+                return jsRuntimeBase.ObjectRefManager;
+            }
+        }
+
+        public long TrackObject(IDotNetObjectRef dotNetObjectRef)
+        {
+            var dotNetObjectId = Interlocked.Increment(ref _nextId);
+            _trackedRefsById[dotNetObjectId] = dotNetObjectRef;
+
+            return dotNetObjectId;
+        }
+
+        public object FindDotNetObject(long dotNetObjectId)
+        {
+            return _trackedRefsById.TryGetValue(dotNetObjectId, out var dotNetObjectRef)
+                ? dotNetObjectRef.Value
+                : throw new ArgumentException($"There is no tracked object with id '{dotNetObjectId}'. Perhaps the DotNetObjectRef instance was already disposed.", nameof(dotNetObjectId));
+            
+        }
+
+        /// <summary>
+        /// Stops tracking the specified .NET object reference.
+        /// This overload is typically invoked from JS code via JS interop.
```

Comment:
> This overload is typically invoked from JS code via JS interop.

Nit: Is this still true? I see it *can* be called via [this](https://github.com/aspnet/Extensions/pull/1704/files#diff-5bc9491fb100bce7a35de845877af2b1R268), but I thought the more common flow now was for this to be called via `someObjectRef.Dispose()`.

Also, being extra nitty, "overload" implies there are other overloads, but this seems to be the only one.

```suggestion
        /// This may be invoked either by disposing a DotNetObjectRef in .NET code, or via JS interop by calling "dispose" on the corresponding instance in JavaScript code
```